src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php

```diff
@@ -563,6 +561,19 @@ public function getValidPropertyPaths()
         ];
     }
 
+    public function getValidReadPropertyPaths()
+    {
+        $testCases = $this->getValidWritePropertyPaths();
+
+        // Optional paths can only be read and can't be written to.
+        $testCases[] = [(object) ['foo' => (object) ['firstName' => 'Bernhard']], 'foo.bar?', null];
+        $testCases[] = [(object) ['foo' => (object) ['firstName' => 'Bernhard']], 'foo.bar?.baz?', null];
```

Comment:
What is the behavior for `foo.bar?.baz`?

Reply:
Related to this request: https://github.com/symfony/symfony/pull/34497#discussion_r349293069

I think it's a case we may never match but I kept it as it was not an issue in the initial PR

Reply:
You meant https://github.com/symfony/symfony/pull/34497#discussion_r349920351 ? i.e the `baz` property does not exist, so should throw an `NoSuchPropertyException` instead of returning `null`

Reply:
Trying to sum-up the cases: https://3v4l.org/Xpnii#v8.1.10